Singleton for AudioManager

WiktorK01 opened the pull request:
just the first task of the AudioManager project, which is to implement the singleton. Created an AudioManager prefab as well to be added to every scene once it's complete

WiktorK01 committed:
Singleton

WiktorK01 committed:
Merge branch 'develop' into AudioManagerWiktor

TheStress reviewed (commented):
There is a **dependency**, where the audio manager needs to exist in the scene to use it

I deleted the audio manager prefab from your the scene and used your `TESTmoveforward` script to test it, in the update function I wrote `AudioManager.Instance.PlayAudio("Stab Sound")` and it showed an error that the object does not exist.

I would recommend making a `public static AudioManager GetInstance()` function in the `AudioManager` script and moving the "checking if this is the only instance" logic into that function. The static function allows other scripts to get and use the audio manager like this `AudioManager.GetInstance().PlayAudio("Stab Sound")`

TheStress committed:
added clarifying functions and testing

- added PlayAudio() to AudioManager and testing coding in the TESTmoveforward to demonstrate how the audio manager is going to be used

WiktorK01 committed:
fixed singleton

WiktorK01 committed:
Merge branch 'develop' into AudioZonesWiktor

TheStress reviewed (approved):
Looks good

WiktorK01 committed:
SoundClip Dictionary + PlayAudio() function

created a dictionary for the audio manager to keep track of all the audioclips available in Assets>Resources>Audio. the resources folder is to take advantage of the Resources.LoadAll function. Implemented a "PlayAudio" function that works for both global & spatial audio
